commit c439c8880efe13ab63b082603121675a0be93a84 (refs/changes/78/3578/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2018-03-09T12:02:09-08:00 (1 year, 7 months ago)
    
    TRITON-221 sdc-imgadm import broken for lx images
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/CHANGES.md b/CHANGES.md
index 6aac617..23315cf 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,9 @@
 # IMGAPI changelog
 
+## 4.2.1
+
+- TRITON-221 sdc-imgadm import broken for lx images
+
 ## 4.2.0
 
 - TRITON-98 Add artedi metrics collection to IMGAPI
diff --git a/lib/images.js b/lib/images.js
index 6e21113..6905068 100644
--- a/lib/images.js
+++ b/lib/images.js
@@ -2087,10 +2087,12 @@ function apiAdminImportImage(req, res, callback) {
          * is no origin image) when imgadm < v3.7.4.
          */
         function workaroundImgapi651(next) {
+            var imgadmVer = imgadmVersionFromReq(req);
             if (data.type === 'lx-dataset' &&
                     data.requirements &&
                     data.requirements.min_platform &&
-                    !semverGter(imgadmVersionFromReq(req), '3.7.4')) {
+                    imgadmVer &&
+                    !semverGter(imgadmVer, '3.7.4')) {
                 if (originImage && originImage.requirements &&
                         originImage.requirements.min_platform) {
                     log.info({
diff --git a/package.json b/package.json
index 2a658f2..83ffae4 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgapi",
   "description": "Image API to manage images for SDC 7",
-  "version": "4.2.0",
+  "version": "4.2.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
